### PR TITLE
Don't include debug.h from public header file

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
Minor change

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Please ignore the following nxstyle error:
```
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/nrf52/nrf52_sdc.c:361:2: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/nrf52/nrf52_sdc.c:370:2: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/nrf52/nrf52_sdc.c:381:2: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/nrf52/nrf52_sdc.c:390:2: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/nrf52/nrf52_sdc.c:399:2: error: Mixed case identifier found
```

**Comment by xiaoxiang781216** [comment]
Depends on: https://github.com/apache/incubator-nuttx-apps/pull/726

**Comment by masayuki2009** [comment]
>Please ignore the following nxstyle error:

OK but the CI still failed.


```
====================================================================================
Configuration/Tool: stm32f103-minimum/mcp2515,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
  Enabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
  Building NuttX...
canlib_getbaud.c: In function 'canlib_getbaud':
Error: canlib_getbaud.c:75:7: error: implicit declaration of function 'canerr'; did you mean 'clearerr'? [-Werror=implicit-function-declaration]
   75 |       canerr("CANIOC_GET_BITTIMING failed, errno=%d\n", errno);
      |       ^~~~~~
      |       clearerr
cc1: all warnings being treated as errors
```

**Comment by xiaoxiang781216** [comment]
Depends on: https://github.com/apache/incubator-nuttx-apps/pull/732

**Comment by xiaoxiang781216** [comment]
@masayuki2009 all check execpt nxstyle pass.

**Comment by xiaoxiang781216** [comment]
Ping @masayuki2009 .

File: fs/littlefs/Make.defs [hunk 1]
```diff
@@ -31,11 +31,10 @@ CSRCS += lfs.c lfs_util.c
...
 
-CFLAGS += -DLFS_TRACE=finfo
-CFLAGS += -DLFS_DEBUG=finfo
-CFLAGS += -DLFS_WARN=fwarn
-CFLAGS += -DLFS_ERROR=ferr
-CFLAGS += -DLFS_ASSERT=DEBUGASSERT
+CFLAGS += -DLFS_NO_ASSERT
+CFLAGS += -DLFS_NO_DEBUG
+CFLAGS += -DLFS_NO_ERROR
+CFLAGS += -DLFS_NO_WARN
```
Comment (masayuki2009) [inline fs/littlefs/Make.defs:37]: @xiaoxiang781216 

I think we should create a separate PR for these changes.
What do you think?

Comment (xiaoxiang781216) [reply fs/littlefs/Make.defs]: I take another approach by utilizing LFS_CONFIG hook.

**Comment by masayuki2009** [comment]
LGTM

**Pull request merged**